hoverkite-protocol/src/response.rs

```diff
@@ -89,178 +130,157 @@ impl Report {
             [b'C', charger_connected] => Self::ChargeState {
                 charger_connected: ascii_to_bool(charger_connected)?,
             },
-            _ => return Err(Other(ParseError)),
+            [b'p'] => Self::PowerOff,
+            [c, ..] => return Err(Other(ProtocolError::InvalidCommand(c))),
```

Comment:
Or `MessageTooLong`, if `c == b'p'`. I think the way you did it for `Command` is better.

Reply:
I think that was spotted by adding a test to Command. I have added the appropriate test and fixed this here too.